sklearn/ensemble/_forest.py

```diff
@@ -1146,7 +1147,8 @@ class labels (multi-output problem).
     >>> print(clf.predict([[0, 0, 0, 0]]))
     [1]
     """
-    def __init__(self,
+    @_deprecate_positional_args
+    def __init__(self, *,
                  n_estimators=100,
```

Comment:
I'm surprised that you don't think `n_estimators` in RFC should be possibly positional. You don't think the semantics of `RandomForestClassifier(1000)` are clear? Perhaps not...?

Reply:
I guess I don't find it clear, because we could have criterion first and n_estimators second, and the order to me seems pretty arbitrary. WDYT?

That said, I agree that in terms of common usage, people may be passing n_estimators positionally.

Reply:
Yeah, for me it's more a question of: if someone wrote it with positional arguments, would the meaning be reasonably straightforward?